### PR TITLE
Chore: Fix RuntimeError: ... got Future <..> attached to a different loop

### PR DESCRIPTION
**Pull request by abetatos**
In issue #963 I commented about the RuntimeError when trying to use TestsetGenerator. 

In issue  #681 you discussed that using get_event_loop() was the optimal way to solve the issue but it had a Deprecation warning. 

image

In newer python versions get_event_loop() raises a RuntimeError when no event loop is active. Thus we can avoid creating unnecessary new loops that seem to be the problem that was mentioned in both issues. 

I have modified the code the least possible to make it clear what I intended to, although in this setup runner could be avoided as was proposed in #689 

Setup: 
Ragas version: 0.1.7
Python version: Python 3.10.13

### CODE REVIEW

File: src/ragas/executor.py [hunk 1]
```diff
@@ -96,7 +99,6 @@ def run(self):
             results = self.loop.run_until_complete(self._aresults())
         finally:
             self.results = results
-            self.loop.stop()
```
Comment (jjmachan) [inline src/ragas/executor.py:99]: why did you remove this?

Comment (jjmachan) [reply src/ragas/executor.py:99]: Thanks a lot for putting the effort to fix this issue ❤️ 🙂

Comment (abetatos) [reply src/ragas/executor.py:99]: Because if you use get_event_loop from a jupyter notebook you obtain the loop of the own jupyter, thus closing it might raise some problems. There might be another solution but I thought it was the most straightforward one.

Comment (abetatos) [reply src/ragas/executor.py:99]: Thanks to you for creating such an awesome library!

Comment (rbellamy) [reply src/ragas/executor.py:99]: > Because if you use get_event_loop from a jupyter notebook you obtain the loop of the own jupyter, thus closing it might raise some problems.

Thank you so much for this! 🙇‍♂️

**Review by jjmachan — APPROVED** [review]

File: src/ragas/executor.py (hunk 1 above)
Comment (charlie-ang-collibra) [reply src/ragas/executor.py:99]: Hi @abetatos .  Thanks for submitting the PR.  🙏  .   I tried to follow this issue but my knowledge of `asyncio` isn't very good.
